notebooks/predicting_queries/entity_intent_suggestion.ipynb

```diff
@@ -0,0 +1,348 @@
+    "    search_terms,\n",
+    "    pageview_timestamp,\n",
+    "    ROW_NUMBER() OVER (PARTITION BY action.session_id ORDER BY pageview_timestamp DESC) as hit_n\n",
+    "    FROM (\n",
+    "      SELECT\n",
+    "          geoNetwork.country,\n",
+    "          geoNetwork.region,\n",
+    "          geoNetwork.metro,\n",
+    "          CONCAT(CAST(fullVisitorId AS STRING), CAST(visitId AS STRING)) AS session_id,\n",
+    "          string_agg(LOWER(hits.page.searchKeyword)) as search_terms,\n",
+    "          MIN(TIMESTAMP_SECONDS(visitStartTime+CAST(hits.time/1000 AS INT64))) as first_search_timestamp\n",
+    "          FROM\n",
+    "          `govuk-bigquery-analytics.87773428.ga_sessions_*`,\n",
+    "          UNNEST(hits) AS hits\n",
+    "          WHERE\n",
+    "          _table_suffix BETWEEN FORMAT_DATE('%Y%m%d',DATE_SUB(CURRENT_DATE(), INTERVAL 2 DAY))\n",
+    "              AND FORMAT_DATE('%Y%m%d',DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY))\n",
+    "          AND hits.page.searchKeyword IS NOT NULL\n",
+    "          GROUP BY session_id, geoNetwork.region,geoNetwork.metro, geoNetwork.country\n",
+    "      ) AS action\n",
+    "LEFT JOIN (\n",
+    "    SELECT\n",
+    "        CONCAT(CAST(fullVisitorId AS STRING), CAST(visitId AS STRING)) AS session_id,\n",
+    "        hits.page.pageTitle as pageTitle,\n",
+    "        hits.page.pagePath as pagePath,\n",
+    "        TIMESTAMP_SECONDS(visitStartTime+CAST(hits.time/1000 AS INT64)) as pageview_timestamp,\n",
+    "        FROM\n",
+    "        `govuk-bigquery-analytics.87773428.ga_sessions_*` \n",
+    "        CROSS JOIN UNNEST(hits) AS hits\n",
+    "        WHERE _table_suffix BETWEEN FORMAT_DATE('%Y%m%d',DATE_SUB(CURRENT_DATE(), INTERVAL 2 DAY))\n",
+    "          AND FORMAT_DATE('%Y%m%d',DATE_SUB(CURRENT_DATE(), INTERVAL 1 DAY)) \n",
+    "        AND hits.type = \"PAGE\"\n",
+    "      ) as viewedpages\n",
+    "ON viewedpages.session_id = action.session_id\n",
+    "WHERE pageTitle is not null\n",
+    "AND pageview_timestamp < first_search_timestamp\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#queries_df = pandas_gbq.read_gbq(query)\n",
+    "#queries_df.to_csv(PROCESSED_DATA+'/pagehistoryqueries.csv')\n",
+    "queries_df = pd.read_csv(PROJECT_DIR+'/data/processed/pagehistoryqueries.csv')\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Query + session data\n",
+    "queries_df.head(10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Taxons for every bit of content \n",
+    "#taxons_df =graph.run(\"MATCH (c:Cid)-[r:IS_TAGGED_TO]->(t:Taxon)RETURN c.name as pagePath,t.name as taxon\").to_data_frame()\n",
+    "#taxons_df.to_csv(PROCESSED_DATA+'/taxons.csv')\n",
```

Comment:
Do you need to get this environment variable first? I had to, so maybe add the following to the top of the notebook:
```python
PROCESSED_DATA = os.getenv('DIR_DATA_PROCESSED')
```